source/GameWindow.cpp

```diff
@@ -82,7 +82,11 @@ bool GameWindow::Init(bool headless)
 
 	// When running the integration tests, don't create a window nor an OpenGL context.
 	if(headless)
+#if defined(__linux__) && !SDL_VERSION_ATLEAST(2, 0, 22)
+		setenv("SDL_VIDEODRIVER", "dummy");
+#else
 		SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
```

Comment:
To be consistent with my previous comment, you'd need to do this too:

```suggestion
		if(!getenv("SDL_VIDEODRIVER"))
			SDL_SetHint(SDL_HINT_VIDEODRIVER, "dummy");
```

I tested the `if(!getenv("SDL_VIDEODRIVER"))` in a test program, but not the `SDL_SetHint` since I don't have that version of SDL.